### PR TITLE
[SP-1855]PDI-13759] excel input step when using apache POI streaming …

### PR DESCRIPTION
**Pull request by SergeyTravin**
…does not release file xlsx in case of error

### CODE REVIEW

**Comment by SergeyTravin** [comment]
@mattcasters, @brosander, @mattyb149 Could you review it, please? Thanks!
It is backport for https://github.com/pentaho/pentaho-kettle/pull/1321

**Comment by SergeyTravin** [comment]
@mattcasters, @brosander, @mattyb149 
I would be grateful if you could review the PR before end of the July SP.
Thanks a lot!

**Comment by brosander** [comment]
@SergeyTravin why are we manually catching exceptions and then failing with the message?  If you just add a throws declaration to the method signature, the test will show an error if it gets an unexpected exception

**Comment by SergeyTravin** [comment]
@brosander, I removed unnecessary catch's block and added  throws declaration to the method signature.  It seems pretty good now. Could you process it?

File: engine/src/org/pentaho/di/trans/steps/excelinput/staxpoi/StaxPoiWorkbook.java [hunk 1]
```diff
@@ -125,13 +132,19 @@ public KSheet getSheet( String sheetName ) {
...
+      for ( StaxPoiSheet sheet : openSheetsMap.values() ) {
         sheet.close();
-      } catch ( Exception e ) {
-        e.printStackTrace();
       }
+      if ( opcpkg != null ) {
+        //We should not save change in xlsx because it is input step.
+        opcpkg.revert();
+      }
+    } catch ( IOException e ) {
```
Comment (brosander) [inline engine/src/org/pentaho/di/trans/steps/excelinput/staxpoi/StaxPoiWorkbook.java]: @SergeyTravin Shouldn't we be doing the try/catch inside the for loop so we still try to close all sheets even if one fails to close?

**Comment by SergeyTravin** [comment]
@brosander, updated PR according your notes. Could you look at it again?
Great thanks!

**Comment by SergeyTravin** [comment]
@hudak, @mattcasters, @brosander, @mattyb149, Could you review this backport for https://github.com/pentaho/pentaho-kettle/pull/1321? This issue is included to the July SP that finishes on Friday.
/cc @mbradbury2014, @dkincade

**Comment by SergeyTravin** [comment]
@hudak, @mattcasters, @brosander, @mattyb149, Could you review this backport for #1321? This issue is included to the July SP that finishes on Friday.
/cc @mbradbury2014, @dkincade

**Pull request merged**